Detectors/FIT/FDD/simulation/include/FDDSimulation/Digitizer.h

```diff
@@ -32,49 +37,75 @@ class Digitizer
-  {
-    mIntRecord.bc = bc;
-    mIntRecord.orbit = orbit;
-  }
-  const o2::InteractionRecord& GetInteractionRecord() const { return mIntRecord; }
-  o2::InteractionRecord& GetInteractionRecord(o2::InteractionRecord& src) { return mIntRecord; }
-  void SetInteractionRecord(const o2::InteractionRecord& src) { mIntRecord = src; }
-  uint32_t GetOrbit() const { return mIntRecord.orbit; }
-  uint16_t GetBC() const { return mIntRecord.bc; }
+  void SetInteractionRecord(const o2::InteractionTimeRecord& src) { mIntRecord = src; }
 
   void SetTriggers(o2::fdd::Digit* digit);
   Int_t SimulateLightYield(Int_t pmt, Int_t nPhot);
-  Float_t SimulateTimeCFD(Int_t channel);
+  Float_t SimulateTimeCFD(ChannelBCDataF pulse);
+  Float_t IntegrateCharge(ChannelBCDataF pulse);
 
   void init();
   void finish();
 
-  void setMCLabels(o2::dataformats::MCTruthContainer<o2::fdd::MCLabel>* mclb) { mMCLabels = mclb; }
-
  private:
-  long mEventTime;              // TF (run) timestamp
-  InteractionRecord mIntRecord; // Interaction record (orbit, bc) -> InteractionTimeRecord
-  Int_t mEventID;               // ID of the current event
-  Int_t mSrcID;                 // signal, background or QED
+  static constexpr int BCCacheMin = -1, BCCacheMax = 10, NBC2Cache = 1 + BCCacheMax - BCCacheMin;
+
+  void Pulse(int nphe, int parID, double timeHit, std::array<o2::InteractionRecord, NBC2Cache> const& cachedIR, int nCachedIR, int channel);
+
+  BCCache& setBCCache(const o2::InteractionRecord& ir);
+  BCCache* getBCCache(const o2::InteractionRecord& ir);
+
+  void storeBC(const BCCache& bc,
+               std::vector<o2::fdd::Digit>& digitsBC, std::vector<o2::fdd::ChannelData>& digitsCh,
+               o2::dataformats::MCTruthContainer<o2::fdd::MCLabel>& labels);
+
+  long mEventTime;                      // TF (run) timestamp
+  o2::InteractionTimeRecord mIntRecord; // Interaction record (orbit, bc) -> InteractionTimeRecord
+  Int_t mEventID;                       // ID of the current event
+  Int_t mSrcID;                         // signal, background or QED
+  std::deque<BCCache> mCache;           // cached BCs data
+  o2::fdd::Triggers mTriggers;
 
   DigitizationParameters parameters;
-  o2::dataformats::MCTruthContainer<o2::fdd::MCLabel>* mMCLabels = nullptr;
 
-  std::array<std::vector<Float_t>, DigitizationParameters::mNchannels> mTime; // Charge time series aka analogue signal pulse from PM
-  std::vector<Float_t> mTimeCFD;                                              // Time series for CFD measurement
-  UInt_t mNBins;                                                              // Number of bins in pulse series
-  Float_t mBinSize;                                                           // Time width of the pulse bin - HPTDC resolution
+  ChannelBCDataF mTimeCFD;                         // Time series for CFD measurement
+  const Float_t mBinSize = 25.0 / mNTimeBinsPerBC; // Time width of the pulse bin - HPTDC resolution
```

Comment:
Don't know if this is relevant in your case, but we have more precize value for the BC duration: https://github.com/AliceO2Group/AliceO2/blob/1bf7b84657390006c394eec631c23610279bfade/Common/Constants/include/CommonConstants/LHCConstants.h#L30

Reply:
Looks good, used in the code